### DISCUSSION TITLE
Should I add a `class_name` to my plugin?

### DISCUSSION THREAD

**Discussion by nikableh** [godotengine/godot-proposals#8851] [General Discussions]
I've created a plugin that is a modification of a default `Camera3D` node. It is a camera with some animations already attached to it, it is called `MyCamera3D`.

Everything is fine, and I've added my node to the _Create New Node_ menu with `add_custom_type()`. But now I wonder if I should also add `class_name MyCamera3D` to the node's script.

I think I don't need this and `class_name` exists for other situations. Still, I feel like I'm missing something. I don't want to make it difficult for other people to use my node. Would there be a situation that someone needs my node to have a `class_name`?

**Comment by AThousandShips** [comment]
For support questions and help please turn to the other [community channels](https://godotengine.org/community), this is for discussing improvements to the engine, not a general support forum 🙂

**Comment by nikableh** [comment]
There is no need to be passive-aggressive with the 🙂 emoji. I understand perfectly well if that's the wrong place to post such questions. I've wandered here by accident.

**Reply by AThousandShips** [reply]
I'm not passive aggressive, please don't just assume bad intent... Please consider the [code of conduct](https://godotengine.org/code-of-conduct/)

I literally left a smiley to be disarming and clarify that I wasn't criticizing or attacking you for making a mistake, as text communication online is hard to convey tone in...

**Reply by nikableh** [reply]
Then I'm genuinely sorry... It really is difficult to determine what tone does one have on the internet. 

>please don't just assume bad intent

I don't want to make this personal all of a sudden, but I did try not to assume it for a long time. It's just that every time I see you use this emoji – it's either in the context of someone doing something stupid or something similar. I do understand that you don't mean anything bad with this smiley, but many people would agree that it's also commonly used as a sign of smug, condescension or sarcasm.

It's just that very few people I saw used it as an actual smiley. And considering just how many extremely aggressive people are on the internet nowadays...

Of course, that might only be my own personal experience and many people would disagree with it. Either way, again, sorry for assuming the worst about you 💙

**Reply by AThousandShips** [reply]
I haven't seen that smiley being used on these channels in such a way, so we don't share that experience, but thank you and try to keep that in mind, communication on the internet is hard enough without assuming intent
